rasterio/_warp.pyx

```diff
@@ -286,6 +286,9 @@ def _reproject(
     # We need a src_transform and src_dst in this case. These will
     # be copied to the MEM dataset.
     if dtypes.is_ndarray(source):
+        # Copy array if it is a view.
+        if not source.flags['OWNDATA']:
+            source = source.copy()
```

Comment:
@davidbrochart do you think it may be more appropriate to use `numpy.require()` here instead?

https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.require.html